esmvaltool/cmorizers/data/formatters/datasets/esacci_seaice.py

```diff
@@ -0,0 +1,319 @@
+    cube.coord('latitude').attributes = None
+    cube.coord('longitude').attributes = None
+
+    # add aux coord 'typesi'
+    area_type = AuxCoord([1.0], standard_name='area_type', var_name='type',
+                         long_name='Sea Ice area type')
+    cube.add_aux_coord(area_type)
+
+    # add attribute cell_measures
+#    siconc:cell_measures = "area: areacello"
+#    cube.attributes.update({"cell_meaures": "area: areacello"})
+    cube.attributes.locals['cell_measures'] = 'area: areacello'
+
+    # Fix data type
+    cube.data = cube.core_data().astype('float32')
+
+    # save daily results
+    logger.debug("Saving cube\n%s", cube)
+    logger.debug("Setting time dimension to UNLIMITED while saving!")
+    version = attributes['version']
+    attributes['version'] = f'{version}-{region}'
+    save_variable(cube, cube.var_name,
+                  out_dir, attributes,
+                  unlimited_dimensions=['time'])
+
+    # calculate monthly means
+    cube = monthly_statistics(cube, operator='mean')
+    # Remove monthly statistics aux coordinates
+    cube.remove_coord(cube.coord('month_number'))
+    cube.remove_coord(cube.coord('year'))
+    # save monthly results
+    logger.debug("Saving cube\n%s", cube)
+    logger.debug("Setting time dimension to UNLIMITED while saving!")
+    version = attributes['version']
+    attributes['mip'] = var['mip2']
+    definition = cmor_table.get_variable(var['mip2'], var['short_name'])
+    save_variable(cube, cube.var_name,
+                  out_dir, attributes,
+                  unlimited_dimensions=['time'])
+
+    # create and save areacello
+    # (code adadapted from formatter 'nsidc_g02202_sh.py')
+    _create_areacello(cfg, cube, attributes, out_dir)
+
+    logger.info("Finished CMORizing %s", ', '.join(in_files))
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize ESACCI-SEAICE dataset."""
+    glob_attrs = cfg['attributes']
+
+    logger.info("Starting cmorization for tier%s OBS files: %s",
+                glob_attrs['tier'], glob_attrs['dataset_id'])
+    logger.info("Input data from: %s", in_dir)
+    logger.info("Output will be written to: %s", out_dir)
+    logger.info('CMORizing ESACCI-SEAICE version %s', glob_attrs['version'])
+
+    if start_date is None:
+        start_date = datetime(1991, 1, 1)
+    if end_date is None:
+        end_date = datetime(2020, 12, 31)
+
+    regions = ('NH', 'SH')
+
+    for region in regions:
+        for short_name, var in cfg['variables'].items():
+            if 'short_name' not in var:
+                var['short_name'] = short_name
+            loop_date = start_date
+            daily = True
+            while loop_date <= end_date:
+                filepattern = os.path.join(
+                    in_dir, region,
+                    var['file'].format(year=loop_date.year, region=region)
+                )
+                in_files = glob.glob(filepattern)
+                if not in_files:
```

Comment:
This would work even if only one file is available for the year... I believe it could be improved with a more thorough validation.

Reply:
Yes, and this is intended to be able to e.g. also process incomplete years.